File: examples/java/kora-java-scheduling-jdk/src/main/java/ru/tinkoff/kora/example/scheduling/jdk/FixRateScheduler.java
package ru.tinkoff.kora.example.scheduling.jdk;

import java.time.temporal.ChronoUnit;
import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleAtFixedRate;

@Component
public final class FixRateScheduler {

    private int state = 0;

    @ScheduleAtFixedRate(initialDelay = 50, period = 50, unit = ChronoUnit.MILLIS)
    void schedule() {
        state++;
    }

    public int getState() {
        return state;
    }
}

File: examples/kotlin/kora-kotlin-scheduling-jdk/src/main/kotlin/io/koraframework/kotlin/example/scheduling/jdk/FixRateScheduler.kt
package io.koraframework.kotlin.example.scheduling.jdk

import io.koraframework.common.annotation.Component
import io.koraframework.scheduling.jdk.annotation.ScheduleAtFixedRate
import io.koraframework.scheduling.jdk.annotation.ScheduleOnce
import io.koraframework.scheduling.jdk.annotation.ScheduleWithFixedDelay
import java.time.temporal.ChronoUnit

@Component
class FixRateScheduler {
    var state = 0
        private set

    @ScheduleAtFixedRate(initialDelay = 50, period = 50, unit = ChronoUnit.MILLIS)
    fun schedule() {
        state++
    }
}

